gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -499,9 +502,9 @@ case class TransformPreOverrides(isAdaptiveContextOrTopParentExchange: Boolean)
    */
   def applyScanTransformer(plan: SparkPlan): SparkPlan = plan match {
     case plan: FileSourceScanExec =>
-      val newPartitionFilters = {
-        ExpressionConverter.transformDynamicPruningExpr(plan.partitionFilters)
-      }
+      val newPartitionFilters = ExpressionConverter.transformDynamicPruningExpr(
+        plan.partitionFilters,
+        conf.adaptiveExecutionEnabled && conf.subqueryReuseEnabled)
```

Comment:
As we know, spark has its own logic to check a given plan to determine whether it should go into AQE logic (see `shouldApplyAQE`&`supportAdaptive` in spark code), not just depending on the config for `adaptiveExecutionEnabled`. So previously, we have a hack approach to discover AQE context at runtime, see `isAdaptiveContext` at `ColumnarOverride.scala`. Should we use it to confirm AQE is applied? Thanks!

Reply:
I note `transformDynamicPruningExpr` will also check whether `newIn.child` is `AdaptiveSparkPlanExec`. If this check confirms AQE is applied with no exceptional case, the proposed patch may work well. Thanks!

Reply:
Done, use `isAdaptiveContext` instead of the `adaptiveExecutionEnabled`